File: Easy/PascalsTriangle.py
class Solution:
    def generate(self, numRows):
        
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        
        tri = []
        for row in range(numRows):
            this_row = [None for i in range(row + 1)]
            this_row[0], this_row[-1] = 1, 1
            for i in range(1, len(this_row) - 1):
                this_row[i] = tri[row - 1][i - 1] + tri[row - 1][i]
            tri.append(this_row)
        return tri

File: Java/Easy/PascalsTriangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> triangle = new ArrayList<List<Integer>>();
        
        if(numRows == 0)
            return triangle;
    
        triangle.add(new ArrayList<>());
        triangle.get(0).add(1);
        
        for(int i = 1; i < numRows; i++){
            List<Integer> row = new ArrayList<>();
            List<Integer> lastRow = triangle.get(i - 1);
            row.add(1);
            
            for(int j = 1; j < i; j++){
                row.add(lastRow.get(j - 1) + lastRow.get(j));
            }

            row.add(1);
            triangle.add(row);
        }
        
        return triangle;
    }
}